<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Heterogeneous Multilayer Network for Multi-Omics Single-Cell Data • HuMMuS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Heterogeneous Multilayer Network for Multi-Omics Single-Cell Data">
<meta name="description" content="A general framework to infer regulatory mechanisms from multi-omics single-cell data. HuMMuS package provides functions to build individual networks from different single-cell modalities, then to integrate them into a single-cell heterogeneous multilayer network. The package also provides functions to infer regulatory mechanisms from the heterogeneous multilayer network. HuMMuS is based on a flexible framework that can be adapted to any single-cell modalities. We provide study cases for scRNA+scATAC and scRNA+scATAC+snmC(methylation) that can be completed by PPI. Many regulatory mechanisms can be inferred with HuMMuS, notably classical gene regulatory networks (GRN), but TF-target genes, enhancer-gene interactions, TF-enhancer bindings, or even research of heterogeneous communnities of omics features (e.g. ensemble of peaks, genes and TFs cooperating to the same biological function).">
<meta property="og:description" content="A general framework to infer regulatory mechanisms from multi-omics single-cell data. HuMMuS package provides functions to build individual networks from different single-cell modalities, then to integrate them into a single-cell heterogeneous multilayer network. The package also provides functions to infer regulatory mechanisms from the heterogeneous multilayer network. HuMMuS is based on a flexible framework that can be adapted to any single-cell modalities. We provide study cases for scRNA+scATAC and scRNA+scATAC+snmC(methylation) that can be completed by PPI. Many regulatory mechanisms can be inferred with HuMMuS, notably classical gene regulatory networks (GRN), but TF-target genes, enhancer-gene interactions, TF-enhancer bindings, or even research of heterogeneous communnities of omics features (e.g. ensemble of peaks, genes and TFs cooperating to the same biological function).">
<meta property="og:image" content="https://cantinilab.github.io/HuMMuS/logo.png">
<!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-36XYS9J0B3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-36XYS9J0B3');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">HuMMuS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/add_networks.html">Add external networks to HuMMuS object</a></li>
    <li><a class="dropdown-item" href="articles/chen_vignette.html">Predict GRN from scRNA+scATAC data (Chen 2018 dataset)</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/cantinilab/HuMMuS"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://research.pasteur.fr/en/team/machine-learning-for-integrative-genomics/"><span class="fa fas fa-university"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="hummus-">HuMMuS <a class="anchor" aria-label="anchor" href="#hummus-"></a>
</h1>
</div>
<p><img src="https://github.com/cantinilab/HuMMuS/actions/workflows/pkgdown.yaml/badge.svg" alt="pkgdown"><a href="https://github.com/cantinilab/HuMMuS/actions/workflows/pages/pages-build-deployment?theme=flickr" class="external-link"><img src="https://github.com/cantinilab/HuMMuS/actions/workflows/pages/pages-build-deployment/badge.svg" alt="doc-deployment"></a> <a href="https://img.shields.io/pypi/v/hummuspy?color=pink" class="external-link"><img src="https://img.shields.io/pypi/v/hummuspy?color=blue" alt="PyPI version"></a></p>
<div class="section level3">
<h3 id="heterogeneous-multilayer-network-for-multi-omics-single-cell-data">Heterogeneous Multilayer network for Multi-omics Single-cell data<a class="anchor" aria-label="anchor" href="#heterogeneous-multilayer-network-for-multi-omics-single-cell-data"></a>
</h3>
<p>HuMMuS exploits multi-omics single-cell measurements to infer numerous regulatory mechanisms. Inter-omics (e.g. peak-gene, TF-peak) and intra-omics interactions (e.g. peak-peak, gene-gene, TF-TF) are considered to capture both regulatory interactions and macromolecule cooperations.</p>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The current outputs available from HuMMuS are</p>
<ul>
<li>gene regulatory networks (GRNs)</li>
<li>enhancers</li>
<li>TF - DNA binding regions</li>
<li>TF - target genes.</li>
</ul>
<div class="section level4">
<h4 id="read-our-publication-for-more-details-">
<a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btae143/7625061" class="external-link">Read our publication</a> for more details !<a class="anchor" aria-label="anchor" href="#read-our-publication-for-more-details-"></a>
</h4>
<p><img src="reference/figures/Fig_0001.jpg" align="center" width="1000"></p>
</div>
<div class="section level3">
<h3 id="scrna--scatac">
<strong>scRNA + scATAC</strong><a class="anchor" aria-label="anchor" href="#scrna--scatac"></a>
</h3>
<p>Like most of the current state-of-the-art methods to infer GRN, we propose a minimal version of HuMMuS based on scRNA-seq + scATAC-seq data (paired or <strong>unpaired</strong>).</p>
</div>
<div class="section level3">
<h3 id="use-of-additional-modalities">
<strong>Use of additional modalities</strong><a class="anchor" aria-label="anchor" href="#use-of-additional-modalities"></a>
</h3>
<p>HuMMuS has been developed to be extendable to any additional biological modality of interest. It is then possible to add any additional network to an already <strong>existing modality</strong> (e.g. both prior-knowledge network and data-driven network of genes), or from a <strong>new modality</strong> (e.g. adding epigenetic or proteomic networks). <br><em>For now, such personalisation requires to use directly some hummuspy (python package) functions at the end of the pipeline and write some configuration files manually. It will be simplified soon !</em></p>
</div>
</div>
<div class="section level2">
<h2 id="tutorialsvignettes">Tutorials/Vignettes<a class="anchor" aria-label="anchor" href="#tutorialsvignettes"></a>
</h2>
<ul>
<li>
<a href="https://cantinilab.github.io/HuMMuS/articles/chen_vignette.html"><strong>Infer a gene regulatory network and other outputs from unpaired/paired scRNA+scATAC data</strong></a> shows the application of HuMMuS to the Chen dataset, used in the benchmark of <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btae143/7625061" class="external-link">HuMMuS publication</a>.</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>HuMMuS is for now ready only in R but requires some python dependencies (hummuspy).</p>
<div class="section level3">
<h3 id="hummus-python-dependency">HuMMuS python dependency<a class="anchor" aria-label="anchor" href="#hummus-python-dependency"></a>
</h3>
<p>Python package <strong>hummuspy</strong> should preferably be installed using pip (from the terminal in a conda environment for e.g)</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a>conda create <span class="sc">-</span>n hummuspy_env python</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>conda activate hummuspy_env</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>pip install hummuspy</span></code></pre></div>
<p>Alternatively, you can also install it directly from R using the reticulate package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rstudio.github.io/reticulate/" class="external-link">reticulate</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/py_install.html" class="external-link">py_install</a></span><span class="op">(</span><span class="st">"hummuspy"</span>, envname <span class="op">=</span> <span class="st">"r-reticulate"</span>, method<span class="op">=</span><span class="st">"auto"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="hummus-r-package">HuMMuS R package<a class="anchor" aria-label="anchor" href="#hummus-r-package"></a>
</h3>
<p>Core R package can be installed directly from R:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"cantinilab/HuMMuS"</span>, ref<span class="op">=</span><span class="st">"dev_SeuratV5"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If you only work SeuratV4, you can also use main branch that will soon be deprecated</span></span>
<span><span class="co">#devtools::install_github("cantinilab/HuMMuS")</span></span></code></pre></div>
<p>Before running HuMMuS, if you’re using multiple conda environment you need to make sure to that reticulate points toward the one where hummuspy is installed. You can precise it at the beginning of your code :</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rstudio.github.io/reticulate/" class="external-link">reticulate</a></span><span class="op">)</span></span>
<span><span class="co"># Using a specific conda environment</span></span>
<span><span class="va">envname</span> <span class="op">=</span> <span class="st">"hummuspy_env"</span> <span class="co"># or "r-reticulate" for e.g.</span></span>
<span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/use_python.html" class="external-link">use_condaenv</a></span><span class="op">(</span><span class="va">envname</span>, required <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>For more details on how to setup the reticulate connection, see: <a href="https://rstudio.github.io/reticulate" class="external-link uri">https://rstudio.github.io/reticulate</a></p>
</div>
<div class="section level3">
<h3 id="scatac-processing">scATAC processing<a class="anchor" aria-label="anchor" href="#scatac-processing"></a>
</h3>
<p>To compute directly the scATAC data with HuMMuS, we currently only propose to use <a href="https://cole-trapnell-lab.github.io/cicero-release/docs_m3/" class="external-link">Cicero</a>. It requires the version running with <a href="https://cole-trapnell-lab.github.io/monocle3/" class="external-link">Monocle3</a>. You then need to install both <a href="https://cole-trapnell-lab.github.io/monocle3/docs/installation/" class="external-link">Monocle3</a>, and Cicero:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"cole-trapnell-lab/monocle3"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"cole-trapnell-lab/cicero-release"</span>, ref <span class="op">=</span> <span class="st">"monocle3"</span><span class="op">)</span></span></code></pre></div>
<p><em>If you encounter some troubles with Monocle3 installation, on ubuntu you can try to run: <code>sudo apt-get install libgdal-dev libgeos-dev libproj-dev</code>. You can also go on <a href="https://github.com/cole-trapnell-lab/monocle3/issues" class="external-link">their github page</a> for more help. Having a previous version of Monocle (1 or 2) still in your R session can cause some troubles. If you encounter some even after restarting your R session,, try to <code>remove.packages("monocle")</code> before to reinstall both Monocle<strong>3</strong> and Cicero</em></p>
</div>
</div>
<div class="section level2">
<h2 id="data-accessibility">Data accessibility<a class="anchor" aria-label="anchor" href="#data-accessibility"></a>
</h2>
<p>To reproduce HuMMuS results presented in the manuscript, preprocessed data <a href="https://figshare.com/projects/Molecular_mechanisms_reconstruction_from_single-cell_multi-omics_data_with_HuMMuS/168899" class="external-link">are accessible here</a> <br> For quick tests, the Chen dataset preprocessed is accessible directly through the package as a Seurat object: <code>load(chen_dataset)</code>, along with a subset version <code>load(chen_dataset_subset)</code>.</p>
</div>
<div class="section level2">
<h2 id="cite-us">Cite us<a class="anchor" aria-label="anchor" href="#cite-us"></a>
</h2>
<p>Trimbour R., Deutschmann I. M., Cantini L. Molecular mechanisms reconstruction from single-cell multi-omics data with HuMMuS. Bioinformatics (2024), btae143. doi: <a href="https://doi.org/10.1093/bioinformatics/btae143" class="external-link uri">https://doi.org/10.1093/bioinformatics/btae143</a></p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>AGPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing HuMMuS</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://research.pasteur.fr/en/member/remi-trimbour/" class="external-link">Rémi Trimbour</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-8770-8412" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://research.pasteur.fr/en/member/remi-trimbour/" class="external-link">Rémi Trimbour</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
