<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Compute peak network from scATAC-seq data — compute_atac_peak_network • HuMMuS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Compute peak network from scATAC-seq data — compute_atac_peak_network"><meta name="description" content="This function will create a network from atac data (or in theory any data
wtih peaks coordinates as features).
Different method should be implemented at some point (e.g. RENIN),
for now Cicero is still the reference and only method available"><meta property="og:description" content="This function will create a network from atac data (or in theory any data
wtih peaks coordinates as features).
Different method should be implemented at some point (e.g. RENIN),
for now Cicero is still the reference and only method available"><meta property="og:image" content="https://cantinilab.github.io/HuMMuS/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-36XYS9J0B3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-36XYS9J0B3');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HuMMuS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/add_networks.html">Add external networks to HuMMuS object</a></li>
    <li><a class="dropdown-item" href="../articles/chen_vignette.html">Predict GRN from scRNA+scATAC data (Chen 2018 dataset)</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/cantinilab/HuMMuS"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://research.pasteur.fr/en/team/machine-learning-for-integrative-genomics/"><span class="fa fas fa-university"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Compute peak network from scATAC-seq data</h1>

      <div class="d-none name"><code>compute_atac_peak_network.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function will create a network from atac data (or in theory any data
wtih peaks coordinates as features).
Different method should be implemented at some point (e.g. RENIN),
for now Cicero is still the reference and only method available</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">compute_atac_peak_network</span><span class="op">(</span></span>
<span>  <span class="va">hummus</span>,</span>
<span>  atac_assay <span class="op">=</span> <span class="st">"peaks"</span>,</span>
<span>  genome <span class="op">=</span> <span class="fu">BSgenome.Hsapiens.UCSC.hg38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.UCSC.hg38/man/package.html" class="external-link">BSgenome.Hsapiens.UCSC.hg38</a></span>,</span>
<span>  method <span class="op">=</span> <span class="st">"cicero"</span>,</span>
<span>  multiplex_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  network_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  store_network <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  number_cells_per_clusters <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  sample_num <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">2025</span>,</span>
<span>  verbose <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  window <span class="op">=</span> <span class="fl">5e+05</span>,</span>
<span>  reduction_method <span class="op">=</span> <span class="st">"UMAP"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-hummus">hummus<a class="anchor" aria-label="anchor" href="#arg-hummus"></a></dt>
<dd><p>(Hummus_Object) - Hummus object</p></dd>


<dt id="arg-atac-assay">atac_assay<a class="anchor" aria-label="anchor" href="#arg-atac-assay"></a></dt>
<dd><p>(character) - Name of the assay containing the atac
peaks data.</p></dd>


<dt id="arg-genome">genome<a class="anchor" aria-label="anchor" href="#arg-genome"></a></dt>
<dd><p>(BSgenome) - Genome used to compute the distance between peaks.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>(character) - Method used to infer network edges.</p><ul><li><p><code>'cicero'</code> - Use cicero to infer regulatory networks.</p></li>
<li><p><code>'Other method'</code> - TO DO.</p></li>
</ul></dd>


<dt id="arg-multiplex-name">multiplex_name<a class="anchor" aria-label="anchor" href="#arg-multiplex-name"></a></dt>
<dd><p>(character) - Name of the multiplex to add the network
to. Default is <code>'peaks'</code>.</p></dd>


<dt id="arg-network-name">network_name<a class="anchor" aria-label="anchor" href="#arg-network-name"></a></dt>
<dd><p>(character) - Name of the network in the multiplex to
add the network to. Default is <code>'peak_network'</code>.</p></dd>


<dt id="arg-store-network">store_network<a class="anchor" aria-label="anchor" href="#arg-store-network"></a></dt>
<dd><p>(bool) - Save the network directly (<code>TRUE</code>,
default) or return without saving on disk (<code>FALSE</code>).</p></dd>


<dt id="arg-output-file">output_file<a class="anchor" aria-label="anchor" href="#arg-output-file"></a></dt>
<dd><p>(character) - Name of the output_file
(if store_network == <code>TRUE</code>).</p></dd>


<dt id="arg-threshold">threshold<a class="anchor" aria-label="anchor" href="#arg-threshold"></a></dt>
<dd><p>(interger, default 0) - Minimal threshold to select tf-gene
edges.</p></dd>


<dt id="arg-number-cells-per-clusters">number_cells_per_clusters<a class="anchor" aria-label="anchor" href="#arg-number-cells-per-clusters"></a></dt>
<dd><p>(integer) - Number of cells grouped by
territory to define pseudocells</p></dd>


<dt id="arg-sample-num">sample_num<a class="anchor" aria-label="anchor" href="#arg-sample-num"></a></dt>
<dd><p>(integer | Cicero) - Number of pseudocells to sample from
each territory. Default is 100.</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>(integer | Cicero) - Seed used to sample pseudocells. Default is
2025</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>(integer) - Display function messages. Set to 0 for no
message displayed, &gt;= 1 for more details.</p></dd>


<dt id="arg-window">window<a class="anchor" aria-label="anchor" href="#arg-window"></a></dt>
<dd><p>(integer) - Size of window to consider potential
cis-regulatory cooperations between peaks. Default is 500K base pairs.</p></dd>


<dt id="arg-reduction-method">reduction_method<a class="anchor" aria-label="anchor" href="#arg-reduction-method"></a></dt>
<dd><p>(character | Cicero) - Method used to reduce dimensionality
of the data to identify territories. Default is <code>'UMAP'</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>(data.frame) - Return list of network interactions between peaks</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Method descriptions :</p><ol><li><p>Cicero
Use patial corelation between peaks that are in a given window (e.g. :
less distant than 500K base pairs)</p></li>
</ol></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu">compute_atac_peak_network</span><span class="op">(</span><span class="va">hummus</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'hummus' not found</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://research.pasteur.fr/en/member/remi-trimbour/" class="external-link">Rémi Trimbour</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

