<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Compute peak network from scATAC-seq data — compute_atac_peak_network • HuMMuS</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Compute peak network from scATAC-seq data — compute_atac_peak_network"><meta property="og:description" content="This function will create a network from atac data (or in theory any data
wtih peaks coordinates as features).
Different method should be implemented at some point (e.g. RENIN),
for now Cicero is still the reference and only method available"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HuMMuS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/chen_grn.html">UNKNOWN TITLE</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Compute peak network from scATAC-seq data</h1>
    
    <div class="hidden name"><code>compute_atac_peak_network.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function will create a network from atac data (or in theory any data
wtih peaks coordinates as features).
Different method should be implemented at some point (e.g. RENIN),
for now Cicero is still the reference and only method available</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">compute_atac_peak_network</span><span class="op">(</span></span>
<span>  <span class="va">hummus</span>,</span>
<span>  atac_assay <span class="op">=</span> <span class="st">"peaks"</span>,</span>
<span>  genome <span class="op">=</span> <span class="fu">BSgenome.Hsapiens.UCSC.hg38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.UCSC.hg38/man/package.html" class="external-link">BSgenome.Hsapiens.UCSC.hg38</a></span>,</span>
<span>  method <span class="op">=</span> <span class="st">"cicero"</span>,</span>
<span>  multiplex_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  network_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  store_network <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output_file <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  number_cells_per_clusters <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  sample_num <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">2025</span>,</span>
<span>  verbose <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  window <span class="op">=</span> <span class="fl">5e+05</span>,</span>
<span>  reduction_method <span class="op">=</span> <span class="st">"UMAP"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>hummus</dt>
<dd><p>(hummus_object) - Hummus object</p></dd>


<dt>atac_assay</dt>
<dd><p>(character) - Name of the assay containing the atac
peaks data.</p></dd>


<dt>genome</dt>
<dd><p>(BSgenome) - Genome used to compute the distance between peaks.</p></dd>


<dt>method</dt>
<dd><p>(character) - Method used to infer network edges.</p><ul><li><p><code>'cicero'</code> - Use cicero to infer regulatory networks.</p></li>
<li><p><code>'Other method'</code> - TO DO.</p></li>
</ul></dd>


<dt>multiplex_name</dt>
<dd><p>(character) - Name of the multiplex to add the network
to. Default is <code>'peaks'</code>.</p></dd>


<dt>network_name</dt>
<dd><p>(character) - Name of the network in the multiplex to
add the network to. Default is <code>'peak_network'</code>.</p></dd>


<dt>store_network</dt>
<dd><p>(bool) - Save the network directly (<code>TRUE</code>,
default) or return without saving on disk (<code>FALSE</code>).</p></dd>


<dt>output_file</dt>
<dd><p>(character) - Name of the output_file
(if store_network == <code>TRUE</code>).</p></dd>


<dt>threshold</dt>
<dd><p>(interger, default 0) - Minimal threshold to select tf-gene
edges.</p></dd>


<dt>number_cells_per_clusters</dt>
<dd><p>(integer) - Number of cells grouped by
territory to define pseudocells</p></dd>


<dt>sample_num</dt>
<dd><p>(integer | Cicero) - Number of pseudocells to sample from
each territory. Default is 100.</p></dd>


<dt>seed</dt>
<dd><p>(integer | Cicero) - Seed used to sample pseudocells. Default is
2025</p></dd>


<dt>verbose</dt>
<dd><p>(integer) - Display function messages. Set to 0 for no
message displayed, &gt;= 1 for more details.</p></dd>


<dt>window</dt>
<dd><p>(integer) - Size of window to consider potential
cis-regulatory cooperations between peaks. Default is 500K base pairs.</p></dd>


<dt>reduction_method</dt>
<dd><p>(character | Cicero) - Method used to reduce dimensionality
of the data to identify territories. Default is <code>'UMAP'</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>(data.frame) - Return list of network interactions between peaks</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Method descriptions :</p><ol><li><p>Cicero
Use patial corelation between peaks that are in a given window (e.g. :
less distant than 500K base pairs)</p></li>
</ol></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu">compute_atac_peak_network</span><span class="op">(</span><span class="va">hummus</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in compute_atac_peak_network(hummus):</span> Please install cicero.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> https://cole-trapnell-lab.github.io/cicero-release/docs_m3/</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Rémi Trimbour.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

