<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference • HuMMuS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Function reference"><meta property="og:image" content="https://cantinilab.github.io/HuMMuS/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-36XYS9J0B3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-36XYS9J0B3');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">HuMMuS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/chen_grn.html">Predict GRN from scRNA+scATAC data (Chen 2018 dataset)</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://saezlab.org/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>
      
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="add_network.html">add_network()</a></code> 
        </dt>
        <dd>Add a network to a multiplex, a multilayer or an hummus object</dd>
      </dl><dl><dt>
          
          <code><a href="aggregate_matrix.html">aggregate_matrix()</a></code> 
        </dt>
        <dd>Aggregate matrix over groups</dd>
      </dl><dl><dt>
          
          <code><a href="bipartite-class.html">bipartite-class</a></code> <code><a href="bipartite-class.html">bipartite</a></code> 
        </dt>
        <dd>Bipartite class</dd>
      </dl><dl><dt>
          
          <code><a href="bipartite_peaks2genes.html">bipartite_peaks2genes()</a></code> 
        </dt>
        <dd>Compute links between DNA regions and genenames</dd>
      </dl><dl><dt>
          
          <code><a href="bipartite_tfs2peaks.html">bipartite_tfs2peaks()</a></code> 
        </dt>
        <dd>Compute links between TFs and DNA regions (ATAC peaks)</dd>
      </dl><dl><dt>
          
          <code><a href="compute_atac_peak_network.html">compute_atac_peak_network()</a></code> 
        </dt>
        <dd>Compute peak network from scATAC-seq data</dd>
      </dl><dl><dt>
          
          <code><a href="compute_gene_network.html">compute_gene_network()</a></code> 
        </dt>
        <dd>Compute gene netwok from scRNA-seq data</dd>
      </dl><dl><dt>
          
          <code><a href="compute_tf_network.html">compute_tf_network()</a></code> 
        </dt>
        <dd>Compute TF network and add it to hummus object</dd>
      </dl><dl><dt>
          
          <code><a href="dMcast.html">dMcast()</a></code> 
        </dt>
        <dd>Copy of the dMcast function from the Matrix.utils package,
since this is off CRAN and does not seem to be maintained anymore
internally</dd>
      </dl><dl><dt>
          
          <code><a href="define_binding_regions.html">define_binding_regions()</a></code> 
        </dt>
        <dd>Define binding_regions from hummus object</dd>
      </dl><dl><dt>
          
          <code><a href="define_enhancers.html">define_enhancers()</a></code> 
        </dt>
        <dd>Define enhancers from hummus object</dd>
      </dl><dl><dt>
          
          <code><a href="define_grn.html">define_grn()</a></code> 
        </dt>
        <dd>Define GRN from hummus object</dd>
      </dl><dl><dt>
          
          <code><a href="define_output.html">define_output()</a></code> 
        </dt>
        <dd>Define output from hummus object</dd>
      </dl><dl><dt>
          
          <code><a href="define_target_genes.html">define_target_genes()</a></code> 
        </dt>
        <dd>Define target genes from hummus object</dd>
      </dl><dl><dt>
          
          <code><a href="fast_aggregate.html">fast_aggregate()</a></code> 
        </dt>
        <dd>Copy of the aggregate.Matrix function from the Matrix.utils package,
since this is off CRAN and does not seem to be maintained anymore
internally</dd>
      </dl><dl><dt>
          
          <code><a href="find_peaks_near_genes.html">find_peaks_near_genes()</a></code> 
        </dt>
        <dd>Associate peaks to genes based on distance to TSS (or gene body)</dd>
      </dl><dl><dt>
          
          <code><a href="format_bipartites_names.html">format_bipartites_names()</a></code> 
        </dt>
        <dd>Format bipartites names for python hummuspy package config functions</dd>
      </dl><dl><dt>
          
          <code><a href="format_multiplex_names.html">format_multiplex_names()</a></code> 
        </dt>
        <dd>Format multiplex names for python hummuspy package config functions</dd>
      </dl><dl><dt>
          
          <code><a href="get_genome_annotations.html">get_genome_annotations()</a></code> 
        </dt>
        <dd>Fetch online genome annotations from Ensembldb database</dd>
      </dl><dl><dt>
          
          <code><a href="get_tf2motifs.html">get_tf2motifs()</a></code> 
        </dt>
        <dd>Fetch online TF motifs from JASPAR2020 and chromVARmotifs</dd>
      </dl><dl><dt>
          
          <code><a href="get_tfs.html">get_tfs()</a></code> 
        </dt>
        <dd>Extract TF names from scRNA data and tf2motifs</dd>
      </dl><dl><dt>
          
          <code><a href="hello.html">hello()</a></code> 
        </dt>
        <dd>Hello, World!</dd>
      </dl><dl><dt>
          
          <code><a href="hummus_object-class.html">hummus_object-class</a></code> <code><a href="hummus_object-class.html">hummus_object</a></code> 
        </dt>
        <dd>The hummus_object class</dd>
      </dl><dl><dt>
          
          <code><a href="motifs_db-class.html">motifs_db-class</a></code> <code><a href="motifs_db-class.html">motifs_db</a></code> 
        </dt>
        <dd>Motifs database class</dd>
      </dl><dl><dt>
          
          <code><a href="multilayer-class.html">multilayer-class</a></code> <code><a href="multilayer-class.html">multilayer</a></code> 
        </dt>
        <dd>Multilayer class</dd>
      </dl><dl><dt>
          
          <code><a href="multiplex-class.html">multiplex-class</a></code> <code><a href="multiplex-class.html">multiplex</a></code> 
        </dt>
        <dd>Multiplex class</dd>
      </dl><dl><dt>
          
          <code><a href="peaks_in_regulatory_elements.html">peaks_in_regulatory_elements()</a></code> 
        </dt>
        <dd>Filter peaks to those overlapping specific (regulatory) elements</dd>
      </dl><dl><dt>
          
          <code><a href="preprocess_cds.html">preprocess_cds()</a></code> 
        </dt>
        <dd>Preprocess a cds to prepare for trajectory inference</dd>
      </dl><dl><dt>
          
          <code><a href="reduce_dimension.html">reduce_dimension()</a></code> 
        </dt>
        <dd>Compute a projection of a cell_data_set object into a lower dimensional
space with non-linear dimension reduction methods</dd>
      </dl><dl><dt>
          
          <code><a href="run_cicero_wrapper.html">run_cicero_wrapper()</a></code> 
        </dt>
        <dd>Cicero wrapper function for the compute_atac_peak_network function</dd>
      </dl><dl><dt>
          
          <code><a href="save_multilayer.html">save_multilayer()</a></code> 
        </dt>
        <dd>Save multilayer object files in a hierarchical structure on disk</dd>
      </dl><dl><dt>
          
          <code><a href="store_network.html">store_network()</a></code> 
        </dt>
        <dd>Wrapper function to save a network or not</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Rémi Trimbour.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

