<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index • HuMMuS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Package index"><meta property="og:image" content="https://cantinilab.github.io/HuMMuS/logo.png"><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-36XYS9J0B3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-36XYS9J0B3');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">HuMMuS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/add_networks.html">Add external networks to HuMMuS object</a></li>
    <li><a class="dropdown-item" href="../articles/chen_vignette.html">Predict GRN from scRNA+scATAC data (Chen 2018 dataset)</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/cantinilab/HuMMuS"><span class="fa fab fa-github"></span></a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://research.pasteur.fr/en/team/machine-learning-for-integrative-genomics/"><span class="fa fas fa-university"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="DefaultAssay.html">DefaultAssay()</a></code> <code><a href="DefaultAssay.html">`DefaultAssay&lt;-`()</a></code>

        </dt>
        <dd>Get Default assays of Hummus_Object (based on Seurat)</dd>
      </dl><dl><dt>

          <code><a href="Hummus_Object-class.html">Hummus_Object-class</a></code> <code><a href="Hummus_Object-class.html">Hummus_Object</a></code>

        </dt>
        <dd>The Hummus_Object class</dd>
      </dl><dl><dt>

          <code><a href="Initiate_Hummus_Object.html">Initiate_Hummus_Object()</a></code>

        </dt>
        <dd>Initiate a hummus object</dd>
      </dl><dl><dt>

          <code><a href="VariableFeatures.html">VariableFeatures()</a></code> <code><a href="VariableFeatures.html">`VariableFeatures&lt;-`()</a></code>

        </dt>
        <dd>Variable features of assays in Hummus_Object (based on Seurat)</dd>
      </dl><dl><dt>

          <code><a href="add_network.html">add_network()</a></code>

        </dt>
        <dd>Add a network to a multiplex, a multilayer or an hummus object</dd>
      </dl><dl><dt>

          <code><a href="aggregate_matrix.html">aggregate_matrix()</a></code>

        </dt>
        <dd>Aggregate matrix over groups</dd>
      </dl><dl><dt>

          <code><a href="bipartite-class.html">bipartite-class</a></code> <code><a href="bipartite-class.html">bipartite</a></code>

        </dt>
        <dd>Bipartite class</dd>
      </dl><dl><dt>

          <code><a href="bipartite_peaks2genes.html">bipartite_peaks2genes()</a></code>

        </dt>
        <dd>Compute links between DNA regions and genenames</dd>
      </dl><dl><dt>

          <code><a href="bipartite_tfs2peaks.html">bipartite_tfs2peaks()</a></code>

        </dt>
        <dd>Compute links between TFs and DNA regions (ATAC peaks)</dd>
      </dl><dl><dt>

          <code><a href="compute_atac_peak_network.html">compute_atac_peak_network()</a></code>

        </dt>
        <dd>Compute peak network from scATAC-seq data</dd>
      </dl><dl><dt>

          <code><a href="compute_gene_network.html">compute_gene_network()</a></code>

        </dt>
        <dd>Compute gene netwok from scRNA-seq data</dd>
      </dl><dl><dt>

          <code><a href="compute_tf_network.html">compute_tf_network()</a></code>

        </dt>
        <dd>Compute TF network and add it to hummus object</dd>
      </dl><dl><dt>

          <code><a href="dMcast.html">dMcast()</a></code>

        </dt>
        <dd>Copy of the dMcast function from the Matrix.utils package, since this is off CRAN and does not seem to be maintained anymore internally</dd>
      </dl><dl><dt>

          <code><a href="define_binding_regions.html">define_binding_regions()</a></code>

        </dt>
        <dd>Define binding_regions from hummus object</dd>
      </dl><dl><dt>

          <code><a href="define_enhancers.html">define_enhancers()</a></code>

        </dt>
        <dd>Define enhancers from hummus object</dd>
      </dl><dl><dt>

          <code><a href="define_general_config.html">define_general_config()</a></code>

        </dt>
        <dd>Define general config file for hummuspy</dd>
      </dl><dl><dt>

          <code><a href="define_grn.html">define_grn()</a></code>

        </dt>
        <dd>Define GRN from hummus object</dd>
      </dl><dl><dt>

          <code><a href="define_output.html">define_output()</a></code>

        </dt>
        <dd>Define output from hummus object</dd>
      </dl><dl><dt>

          <code><a href="define_target_genes.html">define_target_genes()</a></code>

        </dt>
        <dd>Define target genes from hummus object</dd>
      </dl><dl><dt>

          <code><a href="fast_aggregate.html">fast_aggregate()</a></code>

        </dt>
        <dd>Copy of the aggregate.Matrix function from the Matrix.utils package, since this is off CRAN and does not seem to be maintained anymore internally</dd>
      </dl><dl><dt>

          <code><a href="find_peaks_near_genes.html">find_peaks_near_genes()</a></code>

        </dt>
        <dd>Associate peaks to genes based on distance to TSS (or gene body)</dd>
      </dl><dl><dt>

          <code><a href="format_bipartites_names.html">format_bipartites_names()</a></code>

        </dt>
        <dd>Format bipartites names for python hummuspy package config functions</dd>
      </dl><dl><dt>

          <code><a href="format_multiplex_names.html">format_multiplex_names()</a></code>

        </dt>
        <dd>Format multiplex names for python hummuspy package config functions</dd>
      </dl><dl><dt>

          <code><a href="get_genome_annotations.html">get_genome_annotations()</a></code>

        </dt>
        <dd>Fetch online genome annotations from Ensembldb database</dd>
      </dl><dl><dt>

          <code><a href="get_tf2motifs.html">get_tf2motifs()</a></code>

        </dt>
        <dd>Fetch online TF motifs from JASPAR2020 and chromVARmotifs</dd>
      </dl><dl><dt>

          <code><a href="get_tfs.html">get_tfs()</a></code>

        </dt>
        <dd>Extract TF names from scRNA data and tf2motifs</dd>
      </dl><dl><dt>

          <code><a href="motifs_db-class.html">motifs_db-class</a></code> <code><a href="motifs_db-class.html">motifs_db</a></code>

        </dt>
        <dd>Motifs database class</dd>
      </dl><dl><dt>

          <code><a href="multilayer-class.html">multilayer-class</a></code> <code><a href="multilayer-class.html">multilayer</a></code>

        </dt>
        <dd>Multilayer class</dd>
      </dl><dl><dt>

          <code><a href="multiplex-class.html">multiplex-class</a></code> <code><a href="multiplex-class.html">multiplex</a></code>

        </dt>
        <dd>Multiplex class</dd>
      </dl><dl><dt>

          <code><a href="peaks_in_regulatory_elements.html">peaks_in_regulatory_elements()</a></code>

        </dt>
        <dd>Filter peaks to those overlapping specific (regulatory) elements</dd>
      </dl><dl><dt>

          <code><a href="run_cicero_wrapper.html">run_cicero_wrapper()</a></code>

        </dt>
        <dd>Cicero wrapper function for the compute_atac_peak_network function</dd>
      </dl><dl><dt>

          <code><a href="run_tf_null_wrapper.html">run_tf_null_wrapper()</a></code>

        </dt>
        <dd>tf_null wrapper function for the tf_network function</dd>
      </dl><dl><dt>

          <code><a href="save_multilayer.html">save_multilayer()</a></code>

        </dt>
        <dd>Save multilayer object files in a hierarchical structure on disk</dd>
      </dl><dl><dt>

          <code><a href="store_network.html">store_network()</a></code>

        </dt>
        <dd>Wrapper function to save a network or not</dd>
      </dl><dl><dt>

          <code><a href="sub-subset-Hummus_Object.html">`[[`(<i>&lt;Hummus_Object&gt;</i>)</a></code>

        </dt>
        <dd>Access assays in Hummus_Object (based on Seurat)</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by <a href="https://research.pasteur.fr/en/member/remi-trimbour/" class="external-link">Rémi Trimbour</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

