<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Monocle3 aims to learn how cells transition through a
biological program of gene expression changes in an experiment. Each cell
can be viewed as a point in a high-dimensional space, where each dimension
describes the expression of a different gene. Identifying the program of
gene expression changes is equivalent to learning a trajectory that
the cells follow through this space. However, the more dimensions there are
in the analysis, the harder the trajectory is to learn. Fortunately, many
genes typically co-vary with one another, and so the dimensionality of the
data can be reduced with a wide variety of different algorithms. Monocle3
provides two different algorithms for dimensionality reduction via
reduce_dimension (UMAP and tSNE). The function
reduce_dimension is the second step in the trajectory building
process after preprocess_cds.
UMAP is implemented from the package uwot."><title>Compute a projection of a cell_data_set object into a lower dimensional
space with non-linear dimension reduction methods — reduce_dimension • HuMMuS</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Compute a projection of a cell_data_set object into a lower dimensional
space with non-linear dimension reduction methods — reduce_dimension"><meta property="og:description" content="Monocle3 aims to learn how cells transition through a
biological program of gene expression changes in an experiment. Each cell
can be viewed as a point in a high-dimensional space, where each dimension
describes the expression of a different gene. Identifying the program of
gene expression changes is equivalent to learning a trajectory that
the cells follow through this space. However, the more dimensions there are
in the analysis, the harder the trajectory is to learn. Fortunately, many
genes typically co-vary with one another, and so the dimensionality of the
data can be reduced with a wide variety of different algorithms. Monocle3
provides two different algorithms for dimensionality reduction via
reduce_dimension (UMAP and tSNE). The function
reduce_dimension is the second step in the trajectory building
process after preprocess_cds.
UMAP is implemented from the package uwot."><meta property="og:image" content="https://cantinilab.github.io/HuMMuS/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-36XYS9J0B3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-36XYS9J0B3');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">HuMMuS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/chen_grn.html">Predict GRN from scRNA+scATAC data (Chen 2018 dataset)</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://research.pasteur.fr/en/team/machine-learning-for-integrative-genomics/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Compute a projection of a cell_data_set object into a lower dimensional
space with non-linear dimension reduction methods</h1>
      
      <div class="d-none name"><code>reduce_dimension.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Monocle3 aims to learn how cells transition through a
biological program of gene expression changes in an experiment. Each cell
can be viewed as a point in a high-dimensional space, where each dimension
describes the expression of a different gene. Identifying the program of
gene expression changes is equivalent to learning a <em>trajectory</em> that
the cells follow through this space. However, the more dimensions there are
in the analysis, the harder the trajectory is to learn. Fortunately, many
genes typically co-vary with one another, and so the dimensionality of the
data can be reduced with a wide variety of different algorithms. Monocle3
provides two different algorithms for dimensionality reduction via
<code>reduce_dimension</code> (UMAP and tSNE). The function
<code>reduce_dimension</code> is the second step in the trajectory building
process after <code>preprocess_cds</code>.</p>
<p>UMAP is implemented from the package uwot.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">reduce_dimension</span><span class="op">(</span></span>
<span>  <span class="va">cds</span>,</span>
<span>  max_components <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  reduction_method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UMAP"</span>, <span class="st">"tSNE"</span>, <span class="st">"PCA"</span>, <span class="st">"LSI"</span>, <span class="st">"Aligned"</span><span class="op">)</span>,</span>
<span>  preprocess_method <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  umap.metric <span class="op">=</span> <span class="st">"cosine"</span>,</span>
<span>  umap.min_dist <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  umap.n_neighbors <span class="op">=</span> <span class="fl">15L</span>,</span>
<span>  umap.fast_sgd <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  umap.nn_method <span class="op">=</span> <span class="st">"annoy"</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  build_nn_index <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  nn_control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>cds</dt>
<dd><p>the cell_data_set upon which to perform this operation.</p></dd>


<dt>max_components</dt>
<dd><p>the dimensionality of the reduced space. Default is 2.</p></dd>


<dt>reduction_method</dt>
<dd><p>A character string specifying the algorithm to use
for dimensionality reduction. Currently "UMAP", "tSNE", "PCA", "LSI", and "Aligned"
are supported.</p></dd>


<dt>preprocess_method</dt>
<dd><p>A string indicating the preprocessing method used
on the data. Options are "PCA" and "LSI". Default is "LSI".</p></dd>


<dt>umap.metric</dt>
<dd><p>A string indicating the distance metric to be used when
calculating UMAP. Default is "cosine". See uwot package's
<code>umap</code> for details.</p></dd>


<dt>umap.min_dist</dt>
<dd><p>Numeric indicating the minimum distance to be passed to
UMAP function. Default is 0.1.See uwot package's <code>umap</code>
for details.</p></dd>


<dt>umap.n_neighbors</dt>
<dd><p>Integer indicating the number of neighbors to use
during kNN graph construction. Default is 15L. See uwot package's
<code>umap</code> for details.</p></dd>


<dt>umap.fast_sgd</dt>
<dd><p>Logical indicating whether to use fast SGD. Default is
TRUE. See uwot package's <code>umap</code> for details.</p></dd>


<dt>umap.nn_method</dt>
<dd><p>String indicating the nearest neighbor method to be
used by UMAP. Default is "annoy". See uwot package's
<code>umap</code> for details.</p></dd>


<dt>verbose</dt>
<dd><p>Logical, whether to emit verbose output.</p></dd>


<dt>cores</dt>
<dd><p>Number of cores to use for computing the UMAP.</p></dd>


<dt>build_nn_index</dt>
<dd><p>logical When this argument is set to TRUE,
preprocess_cds builds the nearest neighbor index from the
reduced dimension matrix for later use. Default is FALSE.</p></dd>


<dt>nn_control</dt>
<dd><p>An optional list of parameters used to make the nearest
neighbor index. See the set_nn_control help for detailed information.
The default metric is cosine for reduction_methods PCA, LSI, and Aligned,
and is euclidean for reduction_methods tSNE and UMAP. Note: distances in
tSNE space reflect spatial differences poorly so using nearest neighbors
with it may be meaningless.</p></dd>


<dt>...</dt>
<dd><p>additional arguments to pass to the dimensionality reduction
function.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>an updated cell_data_set object</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>UMAP: McInnes, L, Healy, J, UMAP: Uniform Manifold Approximation
and Projection for Dimension Reduction, ArXiv e-prints 1802.03426, 2018</p>
<p>tSNE: Laurens van der Maaten and Geoffrey Hinton. Visualizing
data using t-SNE. J. Mach. Learn. Res., 9(Nov):2579– 2605, 2008.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span>  <span class="co"># \donttest{</span></span></span>
<span class="r-in"><span>    <span class="va">cell_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>,</span></span>
<span class="r-in"><span>                                         <span class="st">'worm_embryo/worm_embryo_coldata.rds'</span>,</span></span>
<span class="r-in"><span>                                         package<span class="op">=</span><span class="st">'monocle3'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">gene_metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>,</span></span>
<span class="r-in"><span>                                         <span class="st">'worm_embryo/worm_embryo_rowdata.rds'</span>,</span></span>
<span class="r-in"><span>                                         package<span class="op">=</span><span class="st">'monocle3'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">expression_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>,</span></span>
<span class="r-in"><span>                                             <span class="st">'worm_embryo/worm_embryo_expression_matrix.rds'</span>,</span></span>
<span class="r-in"><span>                                             package<span class="op">=</span><span class="st">'monocle3'</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">cds</span> <span class="op">&lt;-</span> <span class="fu">new_cell_data_set</span><span class="op">(</span>expression_data<span class="op">=</span><span class="va">expression_matrix</span>,</span></span>
<span class="r-in"><span>                             cell_metadata<span class="op">=</span><span class="va">cell_metadata</span>,</span></span>
<span class="r-in"><span>                             gene_metadata<span class="op">=</span><span class="va">gene_metadata</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in new_cell_data_set(expression_data = expression_matrix, cell_metadata = cell_metadata,     gene_metadata = gene_metadata):</span> could not find function "new_cell_data_set"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span>    <span class="va">cds</span> <span class="op">&lt;-</span> <span class="fu"><a href="preprocess_cds.html">preprocess_cds</a></span><span class="op">(</span><span class="va">cds</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in size_factors(cds):</span> could not find function "size_factors"</span>
<span class="r-in"><span>    <span class="va">cds</span> <span class="op">&lt;-</span> <span class="fu">reduce_dimension</span><span class="op">(</span><span class="va">cds</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'x' in selecting a method for function 'reducedDims': object 'cds' not found</span>
<span class="r-in"><span>  <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Rémi Trimbour.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

