<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="HuMMuS">
<title>Add external networks to HuMMuS object • HuMMuS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Add external networks to HuMMuS object">
<meta property="og:description" content="HuMMuS">
<meta property="og:image" content="https://cantinilab.github.io/HuMMuS/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-36XYS9J0B3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-36XYS9J0B3');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">HuMMuS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/add_networks.html">Add external networks to HuMMuS object</a>
    <a class="dropdown-item" href="../articles/chen_vignette.html">Predict GRN from scRNA+scATAC data (Chen 2018 dataset)</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/cantinilab/HuMMuS">
    <span class="fab fa fab fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://research.pasteur.fr/en/team/machine-learning-for-integrative-genomics/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Add external networks to HuMMuS object</h1>
                        <h4 data-toc-skip class="author">Trimbour
Rémi</h4>
            
            <h4 data-toc-skip class="date">2023-05-16</h4>
      
      
      <div class="d-none name"><code>add_networks.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="general-plan">General plan<a class="anchor" aria-label="anchor" href="#general-plan"></a>
</h3>
<div class="section level5">
<h5 id="initialisation-of-a-hummus-object">1. Initialisation of a hummus object<a class="anchor" aria-label="anchor" href="#initialisation-of-a-hummus-object"></a>
</h5>
</div>
<div class="section level5">
<h5 id="add-a-new-multiplex">2. Add a new multiplex<a class="anchor" aria-label="anchor" href="#add-a-new-multiplex"></a>
</h5>
</div>
<div class="section level5">
<h5 id="add-a-new-bipartite">3. Add a new bipartite<a class="anchor" aria-label="anchor" href="#add-a-new-bipartite"></a>
</h5>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rstudio.github.io/reticulate/" class="external-link">reticulate</a></span><span class="op">)</span></span>
<span><span class="co"># install python dependency</span></span>
<span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/py_install.html" class="external-link">py_install</a></span><span class="op">(</span><span class="st">"hummuspy"</span>, envname <span class="op">=</span> <span class="st">"r-reticulate"</span>, method<span class="op">=</span><span class="st">"auto"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using virtual environment 'r-reticulate' ...</span></span></code></pre>
<pre><code><span><span class="co">## + /home/runner/.virtualenvs/r-reticulate/bin/python -m pip install --upgrade --no-user hummuspy</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://cantinilab.github.io/HuMMuS/">HuMMuS</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="initialisation-of-hummus-object">1. Initialisation of HuMMuS object<a class="anchor" aria-label="anchor" href="#initialisation-of-hummus-object"></a>
</h2>
<p>HuMMuS R objects are instances developed on top of seurat objects. It
means it’s created from a seurat object and the contained assays can be
accessed the same way.</p>
<p>Additionally, it contains a motifs_db object, providing tf motifs
informations, and a multilayer objects, that will be completed while
going through this tutorial. It will mostly include : - list of
multiplex networks (one per modality) - list of bipartites (one per
connection between layers) <img src="figures/hummus_object_description.png" alt="hummus object schema"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create an hummus object from seurat object</span></span>
<span><span class="co"># Load the Chen dataset, which is a Seurat object containing scRNA-seq and scATAC-seq data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"chen_dataset_subset"</span><span class="op">)</span></span>
<span><span class="va">chen_dataset_subset</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class Seurat </span></span>
<span><span class="co">## 12000 features across 385 samples within 2 assays </span></span>
<span><span class="co">## Active assay: RNA (2000 features, 0 variable features)</span></span>
<span><span class="co">##  2 layers present: counts, data</span></span>
<span><span class="co">##  1 other assay present: peaks</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Initiate_Hummus_Object.html">Initiate_Hummus_Object</a></span><span class="op">(</span><span class="va">chen_dataset_subset</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># wrapper of Signac::GetGRangesFromEnsDb, adapting output to UCSC format</span></span>
<span><span class="va">genome_annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genome_annotations.html">get_genome_annotations</a></span><span class="op">(</span></span>
<span>  ensdb_annotations <span class="op">=</span> <span class="fu">EnsDb.Hsapiens.v86</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/EnsDb.Hsapiens.v86/man/package.html" class="external-link">EnsDb.Hsapiens.v86</a></span><span class="op">)</span></span>
<span><span class="co"># Add annotations to peak assay</span></span>
<span><span class="fu">Signac</span><span class="fu">::</span><span class="fu"><a href="https://stuartlab.org/signac/reference/Annotation.html" class="external-link">Annotation</a></span><span class="op">(</span><span class="va">hummus</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">peaks</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">genome_annotations</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">genome_annotations</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load TF motifs from JASPAR2020 and chromVARmotifs in hummus object</span></span>
<span><span class="va">hummus</span><span class="op">@</span><span class="va">motifs_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_tf2motifs.html">get_tf2motifs</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># by default human motifs</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="add-networks">2. Add networks<a class="anchor" aria-label="anchor" href="#add-networks"></a>
</h2>
<div class="float">
<img src="figures/5_steps.png" alt="Steps to build HuMMuS objects"><div class="figcaption">Steps to build HuMMuS objects</div>
</div>
<div class="section level3">
<h3 id="add-a-new-multiplex-through-a-network">2.1 Add a new multiplex through a network<a class="anchor" aria-label="anchor" href="#add-a-new-multiplex-through-a-network"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># The networks should be imported as 3 columns data.frame: (source, target, weight)</span></span>
<span><span class="va">rna_network_path</span> <span class="op">=</span> <span class="st">"chen_multilayer/multiplex/RNA/RNA_GENIE3.tsv"</span></span>
<span><span class="va">rna_network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">rna_network_path</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">rna_network</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       V1      V2        V3</span></span>
<span><span class="co">## 1  BACH1  COL4A6 0.7388727</span></span>
<span><span class="co">## 2  DDIT3 CCDC173 0.7060954</span></span>
<span><span class="co">## 3  DDIT3  CCT6P1 0.6697768</span></span>
<span><span class="co">## 4  ATF6B  DNASE1 0.6598122</span></span>
<span><span class="co">## 5 ARID3B   ACER3 0.6499825</span></span>
<span><span class="co">## 6   ATF4  ATP8A1 0.6465523</span></span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_network.html">add_network</a></span><span class="op">(</span></span>
<span>    <span class="va">hummus</span>, <span class="va">rna_network</span>,</span>
<span>    multiplex_name <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>    network_name <span class="op">=</span> <span class="st">"GENIE3"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  Creating new multiplex :  RNA</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="add-a-network-to-an-existing-multiplex">2.2 Add a network to an existing multiplex<a class="anchor" aria-label="anchor" href="#add-a-network-to-an-existing-multiplex"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_network.html">add_network</a></span><span class="op">(</span></span>
<span>    <span class="va">hummus</span>, <span class="va">rna_network</span>,</span>
<span>    multiplex_name <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>    network_name <span class="op">=</span> <span class="st">"GENIE3_other"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="add-external-bipartites">3 Add external bipartites<a class="anchor" aria-label="anchor" href="#add-external-bipartites"></a>
</h3>
<p>Bipartites can be 2 or 3 columns data.frame, depending on if weighted
or not. Bipartites not computed through HuMMuS have to be added directly
to the object as below:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Add TF peaks bipartite</span></span>
<span><span class="va">atac_rna_path</span> <span class="op">=</span> <span class="st">"chen_multilayer/bipartite/atac_rna.tsv"</span></span>
<span><span class="va">atac_rna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">atac_rna_path</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">atac_rna</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       V1                        V2</span></span>
<span><span class="co">## 1   CMC1    chr3-28241304-28242015</span></span>
<span><span class="co">## 2 ANGEL2  chr1-213015141-213016006</span></span>
<span><span class="co">## 3  CNNM2 chr10-103053902-103054565</span></span>
<span><span class="co">## 4  DNM1L   chr12-32678989-32680007</span></span>
<span><span class="co">## 5   DHX9  chr1-182838974-182840143</span></span>
<span><span class="co">## 6 B3GAT2    chr6-70955759-70956501</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span><span class="op">@</span><span class="va">multilayer</span><span class="op">@</span><span class="va">bipartites</span><span class="op">[[</span><span class="st">"atac_rna"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">new</span><span class="op">(</span><span class="st">"bipartite"</span>,</span>
<span>                           <span class="st">"network"</span> <span class="op">=</span> <span class="va">atac_rna</span>,</span>
<span>                           <span class="st">"multiplex_left"</span> <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>                           <span class="st">"multiplex_right"</span> <span class="op">=</span> <span class="st">"peaks"</span><span class="op">)</span></span></code></pre></div>
<p>Display HuMMuS object</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span></span></code></pre></div>
<pre><code><span><span class="co">## Hummus object containing a multilayer object :</span></span>
<span><span class="co">## Multilayer network containing  1  bipartite networks and  1  multiplex networks.</span></span>
<span><span class="co">##  </span></span>
<span><span class="co">## - Multiplex names:  RNA </span></span>
<span><span class="co">## - Bipartite names:  atac_rna </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## And a Seurat object :</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## 12000 features across samples within 2 assays </span></span>
<span><span class="co">## Active assay: RNA ( features, 0 variable features)</span></span>
<span><span class="co">##  1 other assay present: peaks</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span><span class="op">@</span><span class="va">multilayer</span><span class="op">@</span><span class="va">multiplex</span></span></code></pre></div>
<pre><code><span><span class="co">## $RNA</span></span>
<span><span class="co">## Multiplex of  2  networks with 2000 features.</span></span>
<span><span class="co">##  Networks names:  GENIE3, GENIE3_other</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://research.pasteur.fr/en/member/remi-trimbour/" class="external-link">Rémi Trimbour</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
