<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="HuMMuS">
<title>Predict GRN from scRNA+scATAC data (Chen 2018 dataset) • HuMMuS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Predict GRN from scRNA+scATAC data (Chen 2018 dataset)">
<meta property="og:description" content="HuMMuS">
<meta property="og:image" content="https://cantinilab.github.io/HuMMuS/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-36XYS9J0B3"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-36XYS9J0B3');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">HuMMuS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/add_networks.html">Add external networks to HuMMuS object</a>
    <a class="dropdown-item" href="../articles/chen_vignette.html">Predict GRN from scRNA+scATAC data (Chen 2018 dataset)</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/cantinilab/HuMMuS">
    <span class="fab fa fab fa-github"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://research.pasteur.fr/en/team/machine-learning-for-integrative-genomics/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Predict GRN from scRNA+scATAC data (Chen 2018 dataset)</h1>
                        <h4 data-toc-skip class="author">Trimbour
Rémi</h4>
            
            <h4 data-toc-skip class="date">2023-05-16</h4>
      
      
      <div class="d-none name"><code>chen_vignette.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="useful-links">Useful links:<a class="anchor" aria-label="anchor" href="#useful-links"></a>
</h3>
<p><em>Paper</em>: <a href="https://www.biorxiv.org/content/10.1101/2023.06.09.543828v1" class="external-link uri">https://www.biorxiv.org/content/10.1101/2023.06.09.543828v1</a></p>
<p><em>Github repo</em>: <a href="https://github.com/cantinilab/HuMMuS" class="external-link uri">https://github.com/cantinilab/HuMMuS</a></p>
<p><em>Documentation</em>: <a href="https://cantinilab.github.io/HuMMuS/" class="uri">https://cantinilab.github.io/HuMMuS/</a></p>
</div>
<div class="section level2">
<h2 id="general-description-of-the-pipeline">General description of the pipeline<a class="anchor" aria-label="anchor" href="#general-description-of-the-pipeline"></a>
</h2>
<div class="float">
<img src="figures/schema_HuMMuS.png" alt="Overall pipeline"><div class="figcaption">Overall pipeline</div>
</div>
<div class="section level3">
<h3 id="useful-ressources">Useful ressources<a class="anchor" aria-label="anchor" href="#useful-ressources"></a>
</h3>
<p><em>Preprint detailing the method:</em> <a href="https://www.biorxiv.org/content/10.1101/2023.06.09.543828v1" class="external-link">https://www.biorxiv.org/content/10.1101/2023.06.09.543828v1</a></p>
<p><em>Github repo detailing the installation:</em> <a href="https://github.com/cantinilab/HuMMuS/tree/main" class="external-link">https://github.com/cantinilab/HuMMuS</a></p>
<p><em>Documentation and vignette:</em> <a href="https://cantinilab.github.io/HuMMuS">https://cantinilab.github.io/HuMMuS</a></p>
<div class="section level4">
<h4 id="general-plan">General plan<a class="anchor" aria-label="anchor" href="#general-plan"></a>
</h4>
<div class="section level6">
<h6 id="preparation-of-the-environment">0. Preparation of the environment<a class="anchor" aria-label="anchor" href="#preparation-of-the-environment"></a>
</h6>
</div>
<div class="section level6">
<h6 id="initialisation-of-a-hummus-object">1. Initialisation of a hummus object<a class="anchor" aria-label="anchor" href="#initialisation-of-a-hummus-object"></a>
</h6>
</div>
<div class="section level6">
<h6 id="construction-of-the-multilayer">2. Construction of the multilayer<a class="anchor" aria-label="anchor" href="#construction-of-the-multilayer"></a>
</h6>
</div>
<div class="section level6">
<h6 id="analyse-multilayer-and-define-gene-regulatory-network-grn">3. Analyse multilayer and define gene regulatory network (GRN)<a class="anchor" aria-label="anchor" href="#analyse-multilayer-and-define-gene-regulatory-network-grn"></a>
</h6>
</div>
</div>
</div>
<div class="section level3">
<h3 id="setting-up-the-environment">0. Setting up the environment<a class="anchor" aria-label="anchor" href="#setting-up-the-environment"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rstudio.github.io/reticulate/" class="external-link">reticulate</a></span><span class="op">)</span></span>
<span><span class="co"># install python dependency</span></span>
<span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/py_install.html" class="external-link">py_install</a></span><span class="op">(</span><span class="st">"hummuspy"</span>, envname <span class="op">=</span> <span class="st">"r-reticulate"</span>, method<span class="op">=</span><span class="st">"auto"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using virtual environment 'r-reticulate' ...</span></span></code></pre>
<pre><code><span><span class="co">## + /home/runner/.virtualenvs/r-reticulate/bin/python -m pip install --upgrade --no-user hummuspy</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://cantinilab.github.io/HuMMuS/">HuMMuS</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span></code></pre>
</div>
<div class="section level3">
<h3 id="download-the-single-cell-data">Download the single-cell data<a class="anchor" aria-label="anchor" href="#download-the-single-cell-data"></a>
</h3>
<p>The data used in this tutorial can be <a href="https://figshare.com/account/home#/projects/168899" class="external-link">downloaded
here</a></p>
</div>
<div class="section level3">
<h3 id="initialisation-of-hummus-object">1. Initialisation of HuMMuS object<a class="anchor" aria-label="anchor" href="#initialisation-of-hummus-object"></a>
</h3>
<p>HuMMuS R objects are instances developed on top of seurat objects. It
means it’s created from a seurat object and the contained assays can be
accessed the same way.</p>
<p>Additionally, it contains a motifs_db object, providing tf motifs
informations, and a multilayer objects, that will be completed while
going through this tutorial. It will mostly include :</p>
<ul>
<li>list of multiplex networks (one per modality)</li>
<li>list of bipartites (one per connection between layers) <img src="figures/hummus_object_description.png" alt="hummus object schema">
</li>
</ul>
<div class="section level4">
<h4 id="transform-data-into-a-hummus-object">1.1. Transform data into a hummus object<a class="anchor" aria-label="anchor" href="#transform-data-into-a-hummus-object"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the Chen dataset, which is a Seurat object containing scRNA-seq and scATAC-seq data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"chen_dataset_subset"</span><span class="op">)</span></span>
<span><span class="va">chen_dataset_subset</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class Seurat </span></span>
<span><span class="co">## 12000 features across 385 samples within 2 assays </span></span>
<span><span class="co">## Active assay: RNA (2000 features, 0 variable features)</span></span>
<span><span class="co">##  2 layers present: counts, data</span></span>
<span><span class="co">##  1 other assay present: peaks</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create an hummus object from seurat object</span></span>
<span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Initiate_Hummus_Object.html">Initiate_Hummus_Object</a></span><span class="op">(</span><span class="va">chen_dataset_subset</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="add-genome-and-motif-annotations-to-hummus-object">1.2. Add genome and motif annotations to hummus object<a class="anchor" aria-label="anchor" href="#add-genome-and-motif-annotations-to-hummus-object"></a>
</h4>
<p>Fetch genome annotations online (necessitate an internet connection).
You can also request any “EnsDB” object adapted to your data
(e.g. EnsDb.Hsapiens.v86::EnsDb.Hsapiens.v86 for human genome
annotations) or use your own genome annotations in the same format.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get human genome annotation from EndDb data</span></span>
<span><span class="co"># wrapper of Signac::GetGRangesFromEnsDb, adapting output to UCSC format</span></span>
<span><span class="va">genome_annotations</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genome_annotations.html">get_genome_annotations</a></span><span class="op">(</span></span>
<span>  ensdb_annotations <span class="op">=</span> <span class="fu">EnsDb.Hsapiens.v86</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/EnsDb.Hsapiens.v86/man/package.html" class="external-link">EnsDb.Hsapiens.v86</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># can also be downloaded, saved as an RDS objects for example</span></span></code></pre></div>
<p>Add genome annotations to hummus/seurat object</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Signac</span><span class="fu">::</span><span class="fu"><a href="https://stuartlab.org/signac/reference/Annotation.html" class="external-link">Annotation</a></span><span class="op">(</span><span class="va">hummus</span><span class="op">@</span><span class="va">assays</span><span class="op">$</span><span class="va">peaks</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">genome_annotations</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">genome_annotations</span><span class="op">)</span></span></code></pre></div>
<p>Get TF motifs from JASPAR2020 and chromVARmotifs databsases in a
motifs_db object. By default, human motifs are used. You can specify the
species you want to use with the <code>species</code> argument
(e.g. species = “mouse” for mouse). motifs_db objects contain 3 slots :
* <code>motifs = "PWMatrixList"</code> *
<code>tf2motifs = "data.frame"</code> * <code>tfs = "NULL"</code>
PWMatrixList is a named vector of the motif matrices, whil tf2motifs is
a correspondance table between TFs and motifs. tfs is a named vector of
the TFs. You can also use your own motifs_db object, as long as it
contains the same slots.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load TF motifs from JASPAR2020 and chromVARmotifs in hummus object</span></span>
<span><span class="va">hummus</span><span class="op">@</span><span class="va">motifs_db</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_tf2motifs.html">get_tf2motifs</a></span><span class="op">(</span><span class="op">)</span> <span class="co"># by default human motifs</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="construction-of-the-multilayer-1">2. Construction of the multilayer<a class="anchor" aria-label="anchor" href="#construction-of-the-multilayer-1"></a>
</h3>
<div class="float">
<img src="figures/build_multilayer.png" alt="hummus object schema"><div class="figcaption">hummus object schema</div>
</div>
<p>You can compute the different layers and bipartites as indicated
below.</p>
<p>An example multilayer completed can also be imported with :
<code>data(chen_subset_hummus)</code>. This object corresponds to a
multilayer from chen_dataset_subset completed. You can then go to the
part 3, replacing <code>hummus</code> by <code>chen_subset_hummus</code>
in each step.</p>
<p>Finally, <a href="add_networks.Rmd">you can compute the different
layers before, and add them afterwards</a>. It allows to use faster
methods to compute the networks (e.g. <a href="https://arboreto.readthedocs.io/en/latest/" class="external-link">Arboreto</a> for the
gene network, <a href="https://github.com/r-trimbour/ATACNet" class="external-link">ATACNet</a> for the peak
network, etc.).</p>
<div class="section level4">
<h4 id="compute-3-layers-and-2-bipartites">Compute 3 layers and 2 bipartites<a class="anchor" aria-label="anchor" href="#compute-3-layers-and-2-bipartites"></a>
</h4>
<div class="float">
<img src="figures/5_steps.png" alt="hummus object schema"><div class="figcaption">hummus object schema</div>
</div>
<p><strong>!! Long step !!</strong> You can also go directly to the part
3 for your “discovery tour”. :)</p>
</div>
<div class="section level4">
<h4 id="tf---peaks-bipartite-reconstruction">2.1. TF - peaks bipartite reconstruction<a class="anchor" aria-label="anchor" href="#tf---peaks-bipartite-reconstruction"></a>
</h4>
<p>TF - peaks bipartite is computed using the motifs_db object and the
peak assay. You can specify the assay to use to filter TFs (e.g. “RNA”
if you want to use only the TFs expressed in your dataset). If NULL, all
TFs with motifs will be used. BSGenome object is used to identify
location of motifs and intersect them with peak <br>You can also specify
the name of the bipartite that will be added to the hummus object. By
default, it will be named “tf_peak”.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bipartite_tfs2peaks.html">bipartite_tfs2peaks</a></span><span class="op">(</span></span>
<span>              hummus_object <span class="op">=</span> <span class="va">hummus</span>,</span>
<span>              tf_expr_assay <span class="op">=</span> <span class="st">"RNA"</span>, <span class="co"># use to filter TF on only expressed TFs,</span></span>
<span>                                     <span class="co"># if NULL, all TFs with motifs are used</span></span>
<span>              peak_assay <span class="op">=</span> <span class="st">"peaks"</span>,</span>
<span>              tf_multiplex_name <span class="op">=</span> <span class="st">"TF"</span>,</span>
<span>              genome <span class="op">=</span> <span class="fu">BSgenome.Hsapiens.UCSC.hg38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.UCSC.hg38/man/package.html" class="external-link">BSgenome.Hsapiens.UCSC.hg38</a></span>,</span>
<span>              <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="genes---peaks-bipartite-reconstruction">2.2. Genes - peaks bipartite reconstruction<a class="anchor" aria-label="anchor" href="#genes---peaks-bipartite-reconstruction"></a>
</h4>
<p>Peaks - genes bipartite is computed</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bipartite_peaks2genes.html">bipartite_peaks2genes</a></span><span class="op">(</span></span>
<span>                      hummus_object <span class="op">=</span> <span class="va">hummus</span>,</span>
<span>                      gene_assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>                      peak_assay <span class="op">=</span> <span class="st">"peaks"</span>,</span>
<span>                      store_network <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                      <span class="op">)</span></span></code></pre></div>
<div class="section level5">
<h5 id="compute-layer-networks-and-add-it-to-hummus-object">Compute layer networks and add it to hummus object<a class="anchor" aria-label="anchor" href="#compute-layer-networks-and-add-it-to-hummus-object"></a>
</h5>
<p>Each one of the three layers is computed individually.</p>
</div>
</div>
<div class="section level4">
<h4 id="compute-the-tf-network-from-omnipath-database">2.3. Compute the TF network from OmniPath database<a class="anchor" aria-label="anchor" href="#compute-the-tf-network-from-omnipath-database"></a>
</h4>
<p>We currently use OmniPath R package to fetch TF interactions. You can
first specify if you want to use only the TFs expressed in your dataset
(if you have a RNA assay in your hummus object). If
<code>gene_assay</code> is NULL, all TFs with motifs will be used. <br>
You can then specify which interactions you want to keep through
‘source_target’ argument (“AND” | “OR”). If “AND”, only the interactions
between 2 TFs that are both present in the dataset will be kept. If
“OR”, all interactions involving at least one TF present in the dataset
will be kept. <br>Finally, you can specify the name of the multiplex and
the name of the network that will be added to the hummus object. The
added network will be undirected and unweighted since PPI and OmniPath
database are not directional nor return any weight here.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_tf_network.html">compute_tf_network</a></span><span class="op">(</span><span class="va">hummus</span>,</span>
<span>                            gene_assay <span class="op">=</span> <span class="st">"RNA"</span>, <span class="co"># default = None ;</span></span>
<span>                                                <span class="co"># If a assay is provided,</span></span>
<span>                                                <span class="co"># only the TFs that are present</span></span>
<span>                                                <span class="co"># will be considered</span></span>
<span>                            verbose <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                            <span class="co">#source_target = "OR",</span></span>
<span>                            multiplex_name <span class="op">=</span> <span class="st">"TF"</span>,</span>
<span>                            tf_network_name <span class="op">=</span> <span class="st">"TF_network"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="compute-gene-network-from-scrna-seq-w-genie3">2.4. Compute gene network from scRNA-seq w/ GENIE3<a class="anchor" aria-label="anchor" href="#compute-gene-network-from-scrna-seq-w-genie3"></a>
</h4>
</div>
<div class="section level4">
<h4 id="this-step-can-be-very-slow-if-you-have-thousands-of-cells">
<em>!! This step can be very slow if you have thousands of cells
!!</em><a class="anchor" aria-label="anchor" href="#this-step-can-be-very-slow-if-you-have-thousands-of-cells"></a>
</h4>
<p><em>Current recommendation if you have a big dataset is to compute
the network</em> <em>before with <a href="https://arboreto.readthedocs.io/en/latest/" class="external-link">GRNBoost2 thorugh
arboreto</a></em> <em>and <a href="https://cantinilab.github.io/HuMMuS/articles/add_networks.html">add
it to the hummus object afterwards</a>.</em> Different methods can be
used to compute the gene network. For now, only GENIE3 is implemented in
HuMMuS. You can specify which assay to use to compute the network
(<code>gene_assay</code>). <br>You can specify the number of cores to
use to compute the network. You can also specify if you want to save the
network locally (<code>store_network = TRUE</code>) or not
(<code>store_network = FALSE</code>). If you choose to save the network,
you will need to specify the output file name
(<code>output_file</code>). The returned network will be considered
undirected and weighted. While GENIE3 returns a directed network, we
symmetrize it for the random walk with restart exploration of the genes
proximity.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_gene_network.html">compute_gene_network</a></span><span class="op">(</span></span>
<span>              <span class="va">hummus</span>,</span>
<span>              gene_assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>              method <span class="op">=</span> <span class="st">"GENIE3"</span>,</span>
<span>              verbose <span class="op">=</span> <span class="fl">1</span>,</span>
<span>              number_cores <span class="op">=</span> <span class="fl">5</span>, <span class="co"># GENIE3 method can be ran</span></span>
<span>                                <span class="co"># parallelised on multiple cores</span></span>
<span>              store_network <span class="op">=</span> <span class="cn">FALSE</span>, <span class="co"># by default : FALSE, but</span></span>
<span>                                     <span class="co"># each network can be saved </span></span>
<span>                                     <span class="co"># when computed with hummus</span></span>
<span>              output_file <span class="op">=</span> <span class="st">"gene_network.tsv"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="compute-the-peak-network-from-scatac-seq-w-cicero">2.5. Compute the peak network from scATAC-seq w/ Cicero<a class="anchor" aria-label="anchor" href="#compute-the-peak-network-from-scatac-seq-w-cicero"></a>
</h4>
<p>Different methods can be used to compute the peak network. For now,
only Cicero is implemented in HuMMuS. You can specify which assay to use
to compute the network (<code>peak_assay</code>). You can also specify
the number of cores to use to compute the network. You can also specify
if you want to save the network locally
(<code>store_network = TRUE</code>) or not
(<code>store_network = FALSE</code>). If you choose to save the network,
you will need to specify the output file name
(<code>output_file</code>). The returned network will be considered
undirected and weighted, since cis-regulatory interaction and Cicero
outputs are not directional.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_atac_peak_network.html">compute_atac_peak_network</a></span><span class="op">(</span><span class="va">hummus</span>,</span>
<span>              atac_assay <span class="op">=</span> <span class="st">"peaks"</span>,</span>
<span>              verbose <span class="op">=</span> <span class="fl">1</span>,</span>
<span>              genome <span class="op">=</span> <span class="fu">BSgenome.Hsapiens.UCSC.hg38</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/BSgenome.Hsapiens.UCSC.hg38/man/package.html" class="external-link">BSgenome.Hsapiens.UCSC.hg38</a></span>,</span>
<span>              store_network <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="analyse-of-the-multilayer-and-definition-of-grn">3. Analyse of the multilayer and definition of GRN<a class="anchor" aria-label="anchor" href="#analyse-of-the-multilayer-and-definition-of-grn"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">chen_subset_hummus</span><span class="op">)</span> </span>
<span><span class="va">hummus</span> <span class="op">&lt;-</span> <span class="va">chen_subset_hummus</span></span></code></pre></div>
<div class="section level4">
<h4 id="save-the-mulilayer-in-a-classic-hierarchical-structure">3.1. Save the mulilayer in a classic hierarchical structure<a class="anchor" aria-label="anchor" href="#save-the-mulilayer-in-a-classic-hierarchical-structure"></a>
</h4>
<p>The package used for the random walk with restart exploration
(multixrank) requires currently to save all the network files on disk.
To simplify the organisation of the file, it is possible to save
everything necessary with function <code><a href="../reference/save_multilayer.html">save_multilayer()</a></code>.<br>
It will create a folder (specified through <code>folder_name</code>)
containing all the files necessary to run the multixrank algorithm. The
folder will contain the following subfolders : *
<strong>bipartite</strong> : containing the bipartites files *
<strong>multiplex</strong> : containing the multiplex sub-subfolders *
<strong>multiplex_1</strong> (e.g. TF|peak|RNA) : containing the network
file of each layer of the multiplex * <strong>seed</strong> : that will
contain the seed files (necessary to compute HuMMuS outputs later) *
<strong>config</strong> : that will contain the config files (necessary
to compute HuMMuS outputs later)</p>
<div class="float">
<img src="figures/explore_multilayer.png" alt="hummus object schema"><div class="figcaption">hummus object schema</div>
</div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/save_multilayer.html">save_multilayer</a></span><span class="op">(</span>hummus <span class="op">=</span> <span class="va">hummus</span>,</span>
<span>                folder_name <span class="op">=</span> <span class="st">"chen_multilayer"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in dir.create(folder_name): 'chen_multilayer' already exists</span></span></code></pre>
<pre><code><span><span class="co">## Warning in dir.create(paste0(folder_name, "/", multiplex_folder)):</span></span>
<span><span class="co">## 'chen_multilayer/multiplex' already exists</span></span></code></pre>
<pre><code><span><span class="co">## Warning in dir.create(paste0(folder_name, "/", bipartite_folder)):</span></span>
<span><span class="co">## 'chen_multilayer/bipartite' already exists</span></span></code></pre>
<pre><code><span><span class="co">## Multiplex of  1  networks with 32 features.</span></span>
<span><span class="co">##  Networks names:  TF_network[1] "TF TF_network"</span></span></code></pre>
<pre><code><span><span class="co">## Warning in dir.create(paste0(folder_name, "/", multiplex_folder, "/",</span></span>
<span><span class="co">## multiplex_name)): 'chen_multilayer/multiplex/RNA' already exists</span></span></code></pre>
<pre><code><span><span class="co">## Multiplex of  1  networks with 2000 features.</span></span>
<span><span class="co">##  Networks names:  RNA_GENIE3[1] "RNA RNA_GENIE3"</span></span>
<span><span class="co">## Multiplex of  1  networks with 4639 features.</span></span>
<span><span class="co">##  Networks names:  peak_network_cicero[1] "peaks peak_network_cicero"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="retrieve-target-genes">3.2. Retrieve target genes<a class="anchor" aria-label="anchor" href="#retrieve-target-genes"></a>
</h4>
<p>With HuMMuS, inference of GRN and target gene of TFs are different
outputs. Indeed, while GRN is computed making TFs compete to regulate
genes (by random walk with restart starting from the genes and going to
the TFs), target genes are computed making genes compete to be regulated
by TFs (by random walk with restart starting from the TFs and going to
the genes).<br> For target genes output, you can specify the list of TFs
(<code>tf_list</code>) to use as seed (if NULL by default, all TFs will
be used as seed). Only the links between the seed TFs and the genes will
be computed. You can also specify the list of genes to use. Only the
score of the genes present in the network and the <code>gene_list</code>
will be returned.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ATF2_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/define_target_genes.html">define_target_genes</a></span><span class="op">(</span></span>
<span>  <span class="va">hummus</span>,</span>
<span>  tf_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ATF2"</span><span class="op">)</span>,</span>
<span>  multilayer_f <span class="op">=</span> <span class="st">"chen_multilayer"</span>,</span>
<span>  njobs <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## multiplexes_list :  {'TF': {'TF_network': '00'}, 'RNA': {'RNA_GENIE3': '01'}, 'peaks': {'peak_network_cicero': '01'}}</span></span>
<span><span class="co">## bipartites_list :  {'tf_peak.tsv': {'multiplex_right': 'TF', 'multiplex_left': 'peaks'}, 'atac_rna.tsv': {'multiplex_right': 'peaks', 'multiplex_left': 'RNA'}}</span></span>
<span><span class="co">## folder_multiplexes :  multiplex</span></span>
<span><span class="co">## folder_bipartites :  bipartite</span></span>
<span><span class="co">## gene_list :  None</span></span>
<span><span class="co">## tf_list :  ['ATF2']</span></span>
<span><span class="co">## peak_list :  None</span></span>
<span><span class="co">## config_filename :  target_genes_config.yml</span></span>
<span><span class="co">## config_folder :  config</span></span>
<span><span class="co">## tf_multiplex :  TF</span></span>
<span><span class="co">## peak_multiplex :  peaks</span></span>
<span><span class="co">## rna_multiplex :  RNA</span></span>
<span><span class="co">## update_config :  True</span></span>
<span><span class="co">## save :  False</span></span>
<span><span class="co">## return_df :  True</span></span>
<span><span class="co">## output_f :  None</span></span>
<span><span class="co">## njobs :  1</span></span>
<span><span class="co">## {'TF': {'TF_network': '00'}, 'RNA': {'RNA_GENIE3': '01'}, 'peaks': {'peak_network_cicero': '01'}}</span></span>
<span><span class="co">## bipartites_type has been provided throguh a list, make sure the order matches the one of the 'bipartites' dictionary' keys.</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ATF2_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   layer                   path_layer        score   tf     gene</span></span>
<span><span class="co">## 1   RNA multiplex/RNA/RNA_GENIE3.tsv 5.312726e-05 ATF2    CDC5L</span></span>
<span><span class="co">## 2   RNA multiplex/RNA/RNA_GENIE3.tsv 4.575481e-05 ATF2     ATF3</span></span>
<span><span class="co">## 3   RNA multiplex/RNA/RNA_GENIE3.tsv 4.486195e-05 ATF2    ATP5I</span></span>
<span><span class="co">## 4   RNA multiplex/RNA/RNA_GENIE3.tsv 4.485264e-05 ATF2 CCNB1IP1</span></span>
<span><span class="co">## 5   RNA multiplex/RNA/RNA_GENIE3.tsv 4.482241e-05 ATF2   DGCR14</span></span>
<span><span class="co">## 6   RNA multiplex/RNA/RNA_GENIE3.tsv 4.176625e-05 ATF2    ATAT1</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">target_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/define_target_genes.html">define_target_genes</a></span><span class="op">(</span></span>
<span>  <span class="va">hummus</span>,</span>
<span>  multilayer_f <span class="op">=</span> <span class="st">"chen_multilayer"</span>,</span>
<span>  njobs <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## multiplexes_list :  {'TF': {'TF_network': '00'}, 'RNA': {'RNA_GENIE3': '01'}, 'peaks': {'peak_network_cicero': '01'}}</span></span>
<span><span class="co">## bipartites_list :  {'tf_peak.tsv': {'multiplex_right': 'TF', 'multiplex_left': 'peaks'}, 'atac_rna.tsv': {'multiplex_right': 'peaks', 'multiplex_left': 'RNA'}}</span></span>
<span><span class="co">## folder_multiplexes :  multiplex</span></span>
<span><span class="co">## folder_bipartites :  bipartite</span></span>
<span><span class="co">## gene_list :  None</span></span>
<span><span class="co">## tf_list :  None</span></span>
<span><span class="co">## peak_list :  None</span></span>
<span><span class="co">## config_filename :  target_genes_config.yml</span></span>
<span><span class="co">## config_folder :  config</span></span>
<span><span class="co">## tf_multiplex :  TF</span></span>
<span><span class="co">## peak_multiplex :  peaks</span></span>
<span><span class="co">## rna_multiplex :  RNA</span></span>
<span><span class="co">## update_config :  True</span></span>
<span><span class="co">## save :  False</span></span>
<span><span class="co">## return_df :  True</span></span>
<span><span class="co">## output_f :  None</span></span>
<span><span class="co">## njobs :  1</span></span>
<span><span class="co">## {'TF': {'TF_network': '00'}, 'RNA': {'RNA_GENIE3': '01'}, 'peaks': {'peak_network_cicero': '01'}}</span></span>
<span><span class="co">## bipartites_type has been provided throguh a list, make sure the order matches the one of the 'bipartites' dictionary' keys.</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">target_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   layer                   path_layer        score   tf     gene</span></span>
<span><span class="co">## 1   RNA multiplex/RNA/RNA_GENIE3.tsv 0.0002318538 CUX1   CHRNB1</span></span>
<span><span class="co">## 2   RNA multiplex/RNA/RNA_GENIE3.tsv 0.0002298631 CUX1    CDC5L</span></span>
<span><span class="co">## 3   RNA multiplex/RNA/RNA_GENIE3.tsv 0.0002217500 CUX1    ALG11</span></span>
<span><span class="co">## 4   RNA multiplex/RNA/RNA_GENIE3.tsv 0.0002211191 CUX1    CDCA5</span></span>
<span><span class="co">## 5   RNA multiplex/RNA/RNA_GENIE3.tsv 0.0002201935 CUX1    AGBL5</span></span>
<span><span class="co">## 6   RNA multiplex/RNA/RNA_GENIE3.tsv 0.0001498122 BPTF CDC42SE1</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="define-grn">3.3. Define GRN<a class="anchor" aria-label="anchor" href="#define-grn"></a>
</h4>
<p>The GRN is defined using the multixrank algorithm. It requires to
have the hummuspy python package installed (pip install hummuspy).<br><br> This can be parallelised using the njobs argument. You can also
specify the list of genes and the list of TFs to use.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">grn</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/define_grn.html">define_grn</a></span><span class="op">(</span></span>
<span>  <span class="va">hummus</span>,</span>
<span>  multilayer_f <span class="op">=</span> <span class="st">"chen_multilayer"</span>,</span>
<span>  njobs <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">grn</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="retrieve-enhancers">3.4. Retrieve enhancers<a class="anchor" aria-label="anchor" href="#retrieve-enhancers"></a>
</h4>
<p><br> You can also specify the list of peaks to use.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enhancers</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/define_enhancers.html">define_enhancers</a></span><span class="op">(</span></span>
<span>  <span class="va">hummus</span>,</span>
<span>  gene_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"ATF2"</span><span class="op">)</span>,</span>
<span>  multilayer_f <span class="op">=</span> <span class="st">"chen_multilayer"</span>,</span>
<span>  njobs <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## multiplexes_list :  {'TF': {'TF_network': '00'}, 'RNA': {'RNA_GENIE3': '01'}, 'peaks': {'peak_network_cicero': '01'}}</span></span>
<span><span class="co">## bipartites_list :  {'tf_peak.tsv': {'multiplex_right': 'TF', 'multiplex_left': 'peaks'}, 'atac_rna.tsv': {'multiplex_right': 'peaks', 'multiplex_left': 'RNA'}}</span></span>
<span><span class="co">## folder_multiplexes :  multiplex</span></span>
<span><span class="co">## folder_bipartites :  bipartite</span></span>
<span><span class="co">## gene_list :  ['ATF2']</span></span>
<span><span class="co">## tf_list :  None</span></span>
<span><span class="co">## peak_list :  None</span></span>
<span><span class="co">## config_filename :  enhancers_config.yml</span></span>
<span><span class="co">## config_folder :  config</span></span>
<span><span class="co">## tf_multiplex :  TF</span></span>
<span><span class="co">## peak_multiplex :  peaks</span></span>
<span><span class="co">## rna_multiplex :  RNA</span></span>
<span><span class="co">## update_config :  True</span></span>
<span><span class="co">## save :  False</span></span>
<span><span class="co">## return_df :  True</span></span>
<span><span class="co">## output_f :  None</span></span>
<span><span class="co">## njobs :  1</span></span>
<span><span class="co">## {'TF': {'TF_network': '00'}, 'RNA': {'RNA_GENIE3': '01'}, 'peaks': {'peak_network_cicero': '01'}}</span></span>
<span><span class="co">## bipartites_type has been provided throguh a list, make sure the order matches the one of the 'bipartites' dictionary' keys.</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">enhancers</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   layer                              path_layer        score gene</span></span>
<span><span class="co">## 1 peaks multiplex/peaks/peak_network_cicero.tsv 0.0008960582 ATF2</span></span>
<span><span class="co">## 2 peaks multiplex/peaks/peak_network_cicero.tsv 0.0007440111 ATF2</span></span>
<span><span class="co">## 3 peaks multiplex/peaks/peak_network_cicero.tsv 0.0006710919 ATF2</span></span>
<span><span class="co">## 4 peaks multiplex/peaks/peak_network_cicero.tsv 0.0004391423 ATF2</span></span>
<span><span class="co">## 5 peaks multiplex/peaks/peak_network_cicero.tsv 0.0004031270 ATF2</span></span>
<span><span class="co">## 6 peaks multiplex/peaks/peak_network_cicero.tsv 0.0003824837 ATF2</span></span>
<span><span class="co">##                       peak</span></span>
<span><span class="co">## 1   chr6-44387360-44388305</span></span>
<span><span class="co">## 2 chr3-107522766-107524070</span></span>
<span><span class="co">## 3 chr7-102125135-102125606</span></span>
<span><span class="co">## 4   chr7-73738654-73739271</span></span>
<span><span class="co">## 5  chr10-73874242-73875104</span></span>
<span><span class="co">## 6  chr16-67561885-67563510</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="retrieve-binding-regions">3.5. Retrieve binding regions<a class="anchor" aria-label="anchor" href="#retrieve-binding-regions"></a>
</h4>
<p>For binding regions output, you can specify the list of TFs
(<code>tf_list</code>) to use as seed (if NULL by default, all TFs will
be used as seed). Only the links between the seed TFs and the peaks will
be computed. You can also specify the list of peaks to use. Only the
score of the peaks present in the network and the <code>peak_list</code>
will be returned.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">binding_regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/define_binding_regions.html">define_binding_regions</a></span><span class="op">(</span></span>
<span>  <span class="va">hummus</span>,</span>
<span>  multilayer_f <span class="op">=</span> <span class="st">"chen_multilayer"</span>,</span>
<span>  njobs <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## multiplexes_list :  {'TF': {'TF_network': '00'}, 'RNA': {'RNA_GENIE3': '01'}, 'peaks': {'peak_network_cicero': '01'}}</span></span>
<span><span class="co">## bipartites_list :  {'tf_peak.tsv': {'multiplex_right': 'TF', 'multiplex_left': 'peaks'}, 'atac_rna.tsv': {'multiplex_right': 'peaks', 'multiplex_left': 'RNA'}}</span></span>
<span><span class="co">## folder_multiplexes :  multiplex</span></span>
<span><span class="co">## folder_bipartites :  bipartite</span></span>
<span><span class="co">## gene_list :  None</span></span>
<span><span class="co">## tf_list :  None</span></span>
<span><span class="co">## peak_list :  None</span></span>
<span><span class="co">## config_filename :  binding_regions_config.yml</span></span>
<span><span class="co">## config_folder :  config</span></span>
<span><span class="co">## tf_multiplex :  TF</span></span>
<span><span class="co">## peak_multiplex :  peaks</span></span>
<span><span class="co">## rna_multiplex :  RNA</span></span>
<span><span class="co">## update_config :  True</span></span>
<span><span class="co">## save :  False</span></span>
<span><span class="co">## return_df :  True</span></span>
<span><span class="co">## output_f :  None</span></span>
<span><span class="co">## njobs :  1</span></span>
<span><span class="co">## {'TF': {'TF_network': '00'}, 'RNA': {'RNA_GENIE3': '01'}, 'peaks': {'peak_network_cicero': '01'}}</span></span>
<span><span class="co">## bipartites_type has been provided throguh a list, make sure the order matches the one of the 'bipartites' dictionary' keys.</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">binding_regions</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   layer                              path_layer       score   tf</span></span>
<span><span class="co">## 1 peaks multiplex/peaks/peak_network_cicero.tsv 0.001680402 CUX1</span></span>
<span><span class="co">## 2 peaks multiplex/peaks/peak_network_cicero.tsv 0.001675393 CUX1</span></span>
<span><span class="co">## 3 peaks multiplex/peaks/peak_network_cicero.tsv 0.001674121 CUX1</span></span>
<span><span class="co">## 4 peaks multiplex/peaks/peak_network_cicero.tsv 0.001559648 CUX1</span></span>
<span><span class="co">## 5 peaks multiplex/peaks/peak_network_cicero.tsv 0.001502478 CUX1</span></span>
<span><span class="co">## 6 peaks multiplex/peaks/peak_network_cicero.tsv 0.001502093 CUX1</span></span>
<span><span class="co">##                       peak</span></span>
<span><span class="co">## 1   chr1-65002512-65003322</span></span>
<span><span class="co">## 2   chrX-48971714-48972653</span></span>
<span><span class="co">## 3   chr1-65419692-65420979</span></span>
<span><span class="co">## 4   chrX-48918888-48919598</span></span>
<span><span class="co">## 5   chr8-24955892-24956818</span></span>
<span><span class="co">## 6 chr2-201642365-201643651</span></span></code></pre>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://research.pasteur.fr/en/member/remi-trimbour/" class="external-link">Rémi Trimbour</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
